lib/editors.js

```diff
@@ -0,0 +1,1477 @@
+                    type: 'sync_from_disk',
+                    description: 'Sync course',
+                    job_sequence_id: job_sequence_id,
+                    on_success: _reloadQuestionServers,
+                    on_error: _finishWithFailure,
+                    no_job_sequence_update: true,
+                };
+                serverJobs.createJob(jobOptions, (err, job) => {
+                    if (ERR(err, (e) => logger.error(e))) {
+                        _finishWithFailure();
+                        return;
+                    }
+                    syncFromDisk.syncDiskToSql(this.locals.course.path, this.locals.course.id, job, (err) => {
+                        if (ERR(err, (e) => logger.error(e))) {
+                            job.fail(err);
+                        } else {
+                            job.succeed();
+                        }
+                    });
+                });
+            };
+
+            const _reloadQuestionServers = () => {
+                debug(`${job_sequence_id}: _reloadQuestionServers`);
+                const jobOptions = {
+                    course_id: options.course_id,
+                    user_id: options.user_id,
+                    authn_user_id: options.authn_user_id,
+                    type: 'reload_question_servers',
+                    description: 'Reload server.js code (for v2 questions)',
+                    job_sequence_id: job_sequence_id,
+                    on_success: _finishWithSuccess,
+                    on_error: _finishWithFailure,
+                    no_job_sequence_update: true,
+                };
+                serverJobs.createJob(jobOptions, (err, job) => {
+                    if (ERR(err, (e) => logger.error(e))) {
+                        _finishWithFailure();
+                        return;
+                    }
+                    const coursePath = this.locals.course.path;
+                    requireFrontend.undefQuestionServers(coursePath, job, (err) => {
+                        if (ERR(err, (e) => logger.error(e))) {
+                            job.fail(err);
+                        } else {
+                            job.succeed();
+                        }
+                    });
+                });
+            };
+
+            const _cleanupAfterCommit = (id) => {
+                debug(`Job id ${id} has failed (after git commit)`);
+                jobSequenceHasFailed = true;
+                debug(`${job_sequence_id}: _cleanupAfterCommit`);
+                const jobOptions = {
+                    course_id: options.course_id,
+                    user_id: options.user_id,
+                    authn_user_id: options.authn_user_id,
+                    job_sequence_id: job_sequence_id,
+                    type: 'git_reset',
+                    description: 'Roll back commit',
+                    command: 'git',
+                    arguments: ['reset', '--hard', 'HEAD~1'],
```

Comment:
Should we reset to `origin/master` for safety?

Actually, `_reset()` already does this. Why do we need `_cleanupAfterCommit()`?

Reply:
good call. done.